pkg/operator/ceph/cluster/controller.go

```diff
@@ -388,8 +388,28 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 		return reconcile.Result{}, nil // do not requeue the delete
 	}
 
-	logger.Infof("cleaning up CephCluster %q", nsName)
+	logger.Info("removing networkFence if matching cephCluster UID found")
+	networkFenceList := &addonsv1alpha1.NetworkFenceList{}
+	err := c.client.List(c.OpManagerCtx, networkFenceList)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return reconcile.Result{}, errors.Wrap(err, "failed to list networkFence")
+	}
+	if len(networkFenceList.Items) > 0 {
+		for i := range networkFenceList.Items {
+			if networkFenceList.Items[i].Labels[networkFenceLabel] == string(cluster.GetUID()) {
+				err = opcontroller.RemoveFinalizerWithName(c.OpManagerCtx, c.client, &networkFenceList.Items[i], "csiaddons.openshift.io/network-fence")
+				if err != nil {
+					return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+				}
+				err := c.client.Delete(c.OpManagerCtx, &networkFenceList.Items[i])
```

Comment:
Instead of this Delete for each individual network fence, how about after the `for` loop is done removing finalizers, then you can issue a single `Delete()` that uses the label selector.

Reply:
we do need to pass the object name right?

Reply:
To delete with the label I believe you can call `DeleteAllOf()` and pass the label in the delete options.